Modix.Services/Promotions/PromotionsService.cs

```diff
@@ -245,6 +255,52 @@ public async Task AddCommentAsync(long campaignId, PromotionSentiment sentiment,
             PublishActionNotificationAsync(resultAction);
         }
 
+        public async Task<bool> VotePromoAsync(long campaignId,IGuildUser user, PromotionSentiment sentiment)
+        {
+            await AuthorizationService.OnAuthenticatedAsync(user);
+            AuthorizationService.RequireClaims(AuthorizationClaim.PromotionsComment);
+
+            if (await PromotionCommentRepository.AnyAsync(new PromotionCommentSearchCriteria
+            {
+                CampaignId = campaignId,
+                CreatedById = AuthorizationService.CurrentUserId.Value,
+                IsModified = false
+            }))
+                return false;
+
+            var campaign = await PromotionCampaignRepository.ReadDetailsAsync(campaignId);
+
+            if (campaign.Subject.Id == AuthorizationService.CurrentUserId)
+                return false;
+
+            if (!(campaign.CloseAction is null))
+                return false;
+
+            var rankRoles = await GetRankRolesAsync(AuthorizationService.CurrentGuildId.Value);
+
+            if (!await CheckIfUserIsRankOrHigherAsync(rankRoles, AuthorizationService.CurrentUserId.Value,
+                campaign.TargetRole.Id))
+                return false;
+
+            PromotionActionSummary resultAction;
+
+            using (var transaction = await PromotionCommentRepository.BeginCreateTransactionAsync())
+            {
+                resultAction = await PromotionCommentRepository.CreateAsync(new PromotionCommentCreationData()
+                {
+                    GuildId = campaign.GuildId,
+                    CampaignId = campaignId,
+                    Sentiment = sentiment,
+                    Content = $"This vote to {sentiment.ToString()} was cast on discord",
```

Comment:
What the placeholder comment should be is also up for discussion, i just have it as this but if someone has a better idea

Reply:
I was just figuring we'd refactor to allow it to be `null`, but I suppose this works too. I'd simplify to just something like
`"Vote cast from Discord"`.